src/tesseract.rs

```diff
@@ -74,10 +74,28 @@ impl TessApi {
         }
     }
 
+    /// Provide an image for Tesseract to recognize.
+    /// 
+    /// set_image clears all recognition results, and sets the rectangle to the full image, so it
+    /// may be followed immediately by a `[Self::get_utf8_text]`, and it will automatically perform
+    /// recognition. 
     pub fn set_image(&mut self, img: &leptonica::Pix) {
+        // "Tesseract takes its own copy of the image, so it need not persist until after Recognize"
         unsafe { capi::TessBaseAPISetImage2(self.raw, img.raw as *mut capi::Pix) }
     }
 
+    // I have no idea if this is correct:
```

Comment:
I think it's correct

I used this test to assert it

```rust
let path = std::path::Path::new("tests/di.png");
let img = leptess::leptonica::pix_read(&path).unwrap();
let mut tes = leptess::tesseract::TessApi::new(None, "eng").unwrap();
tes.set_image(&img);
assert_eq!(tes.get_image_dimensions(), Some((442, 852)));
```

It would be good to add a test like this to make sure it doesn't break in the future.
Even better would be to add it as a doc-test, so we can make sure it's correct and there is a usage example.

https://doc.rust-lang.org/rustdoc/documentation-tests.html

Reply:
What I was wondering: is there a safe way to return a reference to the underlying capi::Pix? Because that's what I wanted to do originally. But then I couldn't think of an appropriate lifetime annotation to give the returned reference. Basically the reference only lives as long _until the next image is set with set_image_. I couldn't find a way to express this logic in lifetimes; do you have an idea on this?

Reply:
Something like this should work:

```rust
    /// Get the pix object.
    ///
    /// # Example
    ///
    /// ```
    /// let path = std::path::Path::new("tests/di.png");
    /// let img = leptess::leptonica::pix_read(&path).unwrap();
    /// let mut tes = leptess::tesseract::TessApi::new(None, "eng").unwrap();
    /// tes.set_image(&img);
    /// let leptess::capi::Pix {w, h, ..} = tes.get_pix().unwrap();
    /// assert_eq!((*w, *h), (442, 852));
    /// ```
    pub fn get_pix(&self) -> Option<&capi::Pix> {
        unsafe {
            let pix = capi::TessBaseAPIGetInputImage(self.raw);
            if pix.is_null() {
                None
            } else {
                Some(&*pix)
            }
        }
    }
```

In the example `TessApi` and the Pix object have the same read only lifetime. No one is allowed to change the pix object.

> Basically the reference only lives as long until the next image is set with set_image

Because we've got a reference to TessApi, no one is allowed to run `set_image` (because it requires a `&mut`).

Reply:
Oh you're right! I totally forgot that. Rust saves the day again

Reply:
neat solution @ccouzens :+1:

Reply:
do we want to add this get_pix method to the api as well?